### PR TITLE
Removed "shoulda" gem in favor of "shoulda-matchers" and update

### PR DESCRIPTION
**Pull request by nisevi**
Hi there, my first contribution here 👍.

I just saw a warning when running all tests while configuring the project for development and I thought that it will be cool to have the `shoulda` gem upgraded 🚀.

The warning:

```
warning: BigDecimal.new is deprecated; use BigDecimal() method instead.
```

`shoulda` gem upgraded to version [3.6.0](https://rubygems.org/gems/shoulda/versions/3.6.0)

`shoulda-matchers` gem upgraded to version [3.1.3](https://github.com/thoughtbot/shoulda-matchers/blob/master/NEWS.md#improvements-1) from version [2.8.0](https://github.com/thoughtbot/shoulda-matchers/blob/master/NEWS.md#280)

Please let me know if I need to add more info 👍

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -45,6 +45,16 @@ def call(env)
 require 'shoulda'
 require 'sidekiq/testing'
 
+# The shoulda-matchers gem no longer detects the test framework
+# you're using or mixes itself into that framework automatically.
```
Comment (nisevi) [inline spec/rails_helper.rb]: [3.0.0](https://github.com/thoughtbot/shoulda-matchers/blob/master/NEWS.md#300)

> The gem no longer detects the test framework you're using or mixes itself into that framework automatically. [History](https://github.com/freerange/mocha/commit/049080c673ee3f76e76adc1e1a6122c7869f1648) has [shown](https://github.com/rr/rr/issues/29) that performing any kind of detection is prone to bugs and more complicated than it should be.

File: spec/models/category_spec.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
   it 'validates uniqueness of name' do
     Fabricate(:category)
-    is_expected.to validate_uniqueness_of(:name).scoped_to(:parent_category_id)
+    is_expected.to validate_uniqueness_of(:name).scoped_to(:parent_category_id).case_insensitive
```
Comment (nisevi) [inline spec/models/category_spec.rb]: [Unexpected failure when using case_insensitive and scoped_to](https://github.com/thoughtbot/shoulda-matchers/issues/1092)

Specifically this [comment](https://github.com/thoughtbot/shoulda-matchers/issues/1092#issuecomment-378864653)

File: Gemfile [hunk 3]
```diff
@@ -135,7 +135,7 @@ group :test, :development do
   gem 'rb-fsevent', require: RUBY_PLATFORM =~ /darwin/i ? 'rb-fsevent' : false
   gem 'rb-inotify', '~> 0.9', require: RUBY_PLATFORM =~ /linux/i ? 'rb-inotify' : false
   gem 'rspec-rails', require: false
-  gem 'shoulda', require: false
+  gem 'shoulda', '~> 3.6'
```
Comment (SamSaffron) [inline Gemfile]: small one, is there a reason you removed the require: false, I think we only need this in test no?

Comment (nisevi) [reply Gemfile]: hmmmmmm you are right! I can't remember the reason. I will update it accordingly 👍

**Comment by nisevi** [comment]
@SamSaffron I have removed `shoulda` gem in favour of `shoulda-matchers` only and I have applied the changes we talked about `require: false` 👍

**Comment by SamSaffron** [comment]
excellent! thanks @nisevi

**Pull request merged**